tools/rewriter/SourceRewriter.cpp

```diff
@@ -1029,15 +1029,20 @@ int main(int argc, const char **argv) {
 
   RefactoringTool tool(options_parser.getCompilations(),
                        options_parser.getSourcePathList());
+
+  // Ensure that exactly one -DIA2_ENABLE is present, defined to zero, so we do
+  // not try to rewrite our own implementation details
   tool.appendArgumentsAdjuster([&](const CommandLineArguments &args, llvm::StringRef filename) {
     CommandLineArguments new_args(args);
     // Try to remove existing definition from command line to avoid warnings.
     new_args.erase(std::remove_if(new_args.begin(), new_args.end(),
                                   [](std::string &x) { return x.starts_with("-DIA2_ENABLE="); }),
                    new_args.end());
-    new_args.push_back("-DIA2_ENABLE=0"s);
+    // Insert prior to the "end of flags" double hyphen; if not present, append
+    auto double_hyphen_pos = std::find(new_args.begin(), new_args.end(), "--");
```

Comment:
Are you sure this vector of arguments includes `--` and the arguments before it? I thought clang just took arguments from the compile commands JSON or those after the `--` instead if it's used.

Reply:
In some situations (e.g. if `-p` is not passed to the writer but a `compile_commands.json` is found and Clang synthesizes an argv from it), this argument vector will contain `--`. In cases where `--` is not present, this still does the right thing.

I really hate Clang's argv handling.